kura/org.eclipse.kura.wire.component.provider/src/main/java/org/eclipse/kura/internal/wire/publisher/CloudPublisher.java

```diff
@@ -54,44 +55,66 @@
  * Payload and will be sent to the Cloud Platform. In addition, the user can
  * avail to wrap every Wire Record as a JSON object as well.
  */
-public final class CloudPublisher implements WireReceiver, DataServiceListener, ConfigurableComponent {
+public final class CloudPublisher implements WireReceiver, CloudClientListener, ConfigurableComponent {
 
-    /** The Cloud Publisher Disconnection Manager. */
-    private static CloudPublisherDisconnectManager disconnectManager;
+    /**
+     * Inner class defined to track the CloudServices as they get added, modified or removed.
+     * Specific methods can refresh the cloudService definition and setup again the Cloud Client.
+     *
+     */
+    private final class CloudPublisherServiceTrackerCustomizer
+            implements ServiceTrackerCustomizer<CloudService, CloudService> {
+
+        @Override
+        public CloudService addingService(ServiceReference<CloudService> reference) {
+            CloudPublisher.this.cloudService = CloudPublisher.this.componentContext.getBundleContext()
+                    .getService(reference);
+            try {
+                // recreate the Cloud Client
+                setupCloudClient();
+            } catch (KuraException e) {
+                s_logger.error(s_message.cloudClientSetupProblem() + ThrowableUtil.stackTraceAsString(e));
+            }
+            return CloudPublisher.this.cloudService;
+        }
+
+        @Override
+        public void modifiedService(ServiceReference<CloudService> reference, CloudService service) {
+            CloudPublisher.this.cloudService = CloudPublisher.this.componentContext.getBundleContext()
+                    .getService(reference);
+            try {
+                // recreate the Cloud Client
+                setupCloudClient();
+            } catch (KuraException e) {
+                s_logger.error(s_message.cloudClientSetupProblem() + ThrowableUtil.stackTraceAsString(e));
+            }
+        }
+
+        @Override
+        public void removedService(ServiceReference<CloudService> reference, CloudService service) {
+            CloudPublisher.this.cloudService = null;
+        }
+    }
 
-    /** The Logger instance. */
```

Comment:
I, personally, don't believe variables declarations should be commented, if the names given are self-explanatory by themselves. For methods, I can agree on having comments, because we usually don't have (as theory requires) simple and plain methods.